monai/data/utils.py

```diff
@@ -712,16 +716,16 @@ def create_file_basename(
     if data_root_dir and filedir:
         filedir_rel_path = os.path.relpath(filedir, data_root_dir)
 
-    # sub-folder path will be original name without the extension
-    subfolder_path = os.path.join(folder_path, filedir_rel_path, filename)
-    if not os.path.exists(subfolder_path):
-        os.makedirs(subfolder_path)
+    # output folder path will be original name without the extension
+    output = os.path.join(folder_path, filedir_rel_path)
 
-    if len(postfix) > 0:
-        # add the sub-folder plus the postfix name to become the file basename in the output path
-        output = os.path.join(subfolder_path, filename + "_" + postfix)
-    else:
-        output = os.path.join(subfolder_path, filename)
+    if separate_folder:
+        output = os.path.join(output, filename)
```

Comment:
not sure if this `filename` should also have the `postfix`?

Reply:
That will be used in the next lines for the subfolders that go in the folder referred to as `output` here.

Reply:
Hi @wyli @ericspod ,

I think you mean to save as: "output/image_seg/image_seg.nii", right? The existing bahvior in MONAI is "output/image/image_seg.nii".
Maybe let's change to "output/image_seg/image_seg.nii"?

Thanks.